File: app/src/main/java/org/tasks/activities/attribution/AttributionRow.java
package org.tasks.activities.attribution;

public class AttributionRow {

  private final boolean isHeader;
  private final String license;
  private final String copyrightHolder;
  private final String libraries;

  AttributionRow(String license) {
    this.license = license;
    isHeader = true;
    copyrightHolder = null;
    libraries = null;
  }

  AttributionRow(String copyrightHolder, String libraries) {
    this.copyrightHolder = copyrightHolder;
    this.libraries = libraries;
    isHeader = false;
    license = null;
  }

  boolean isHeader() {
    return isHeader;
  }

  public String getLicense() {
    return license;
  }

  String getCopyrightHolder() {
    return copyrightHolder;
  }

  public String getLibraries() {
    return libraries;
  }
}

File: app/src/main/java/org/tasks/activities/attribution/AttributionRow.kt
package org.tasks.activities.attribution

class AttributionRow {
    val isHeader: Boolean
    val license: String?
    val copyrightHolder: String?
    val libraries: String?

    internal constructor(license: String?) {
        this.license = license
        isHeader = true
        copyrightHolder = null
        libraries = null
    }

    internal constructor(copyrightHolder: String?, libraries: String?) {
        this.copyrightHolder = copyrightHolder
        this.libraries = libraries
        isHeader = false
        license = null
    }
}